jinahub/encoders/image/TimmImageEncoder/timm_encoder.py

```diff
@@ -0,0 +1,102 @@
+    `Timm library`.
+    https://rwightman.github.io/pytorch-image-models/
+
+    :param model_name: the name of the model. Models listed on:
+        https://rwightman.github.io/pytorch-image-models/models/
+    :param device: Which device the model runs on. Can be 'cpu' or 'cuda'.
+    :param default_traversal_paths: Used in the encode method an defines traversal on the received `DocumentArray`.
+    :param default_batch_size: Defines the batch size for inference on the loaded PyTorch model.
+    :param use_default_preprocessing: Should the input be preprocessed with default configuration.
+    :param args:  Additional positional arguments.
+    :param kwargs: Additional keyword arguments.
+    """
+
+    def __init__(
+        self,
+        model_name: str = "resnet18",
+        device: str = "cpu",
+        default_traversal_path: Tuple = ("r",),
+        default_batch_size: Optional[int] = 32,
+        use_default_preprocessing: bool = True,
+        *args,
+        **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+
+        self.device = device
+        self.default_batch_size = default_batch_size
+        self.use_default_preprocessing = use_default_preprocessing
+
+        self.default_traversal_path = default_traversal_path
+
+        # axis 0 is the batch
+        self._default_channel_axis = 1
+
+        self._model = create_model(model_name, pretrained=True, num_classes=0)
+        self._model.eval()
+
+        config = resolve_data_config({}, model=self._model)
+        self._preprocess = create_transform(**config)
+        self._preprocess.transforms.insert(0, T.ToPILImage())
+
+    @requests
+    def encode(self, docs: Optional[DocumentArray], parameters: Dict, **kwargs):
+        """
+        Encode image data into a ndarray of `D` as dimension, and fill the embedding of each Document.
+
+        :param docs: DocumentArray containing images
+        :param parameters: dictionary to define the `traversal_paths` and the `batch_size`. For example,
+               `parameters={'traversal_paths': ['r'], 'batch_size': 10}`.
+        :param kwargs: Additional key value arguments.
+        """
+        if docs:
+            docs_batch_generator = get_docs_batch_generator(
+                docs,
+                traversal_path=parameters.get(
+                    "traversal_paths", self.default_traversal_path
+                ),
+                batch_size=parameters.get("batch_size", self.default_batch_size),
+                needs_attr="blob",
+            )
+            self._compute_embeddings(docs_batch_generator)
+
+    def _compute_embeddings(self, docs_batch_generator: Iterable) -> None:
+        with torch.no_grad():
+            for document_batch in docs_batch_generator:
+                blob_batch = [d.blob for d in document_batch]
+                if self.use_default_preprocessing:
+                    images = np.stack(self._preprocess_image(blob_batch))
```

Comment:
No point for creating a special function for this one-liner, let's keep it simple